### ISSUE TITLE
driver aborts on transport errors under very likely conditions

### ISSUE THREAD

**Issue by jdef** [mesos/mesos-go#160]
in this case, the mesos master was killed. the driver can't cope with the fact that it can't deliver the message and aborts. not good. it's like that in a production cluster the currently leading mesos master could go down, and there could be messages in the queue already addressed to it. the driver shouldn't abort in this case.

```
I0827 13:37:17.085276    2418 http_transporter.go:343] libproc target URL http://127.0.0.1:5050/master/mesos.internal.KillTaskMessage
I0827 13:37:17.086155    2418 http_transporter.go:80] checking url.Error for recoverability
I0827 13:37:17.086221    2418 http_transporter.go:83] checking net.OpError for recoverability: &net.OpError{Op:"dial", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc8204628d0), Err:(*os.SyscallError)(0xc820011020)}
I0827 13:37:17.086386    2418 http_transporter.go:132] Failed to POST: Post http://127.0.0.1:5050/master/mesos.internal.KillTaskMessage: dial tcp 127.0.0.1:5050: getsockopt: connection refused
I0827 13:37:17.088218    2418 messenger.go:279] Failed to send message mesos.internal.KillTaskMessage: Post http://127.0.0.1:5050/master/mesos.internal.KillTaskMessage: dial tcp 127.0.0.1:5050: getsockopt: connection refused
I0827 13:37:17.088479    2418 messenger.go:343] Receiving message mesos.internal.FrameworkErrorMessage from scheduler(1)@127.0.0.1:50261
I0827 13:37:17.088576    2418 scheduler.go:630] Handling framework error event.
I0827 13:37:17.088606    2418 scheduler.go:1093] Aborting driver, got error ' Failed to send message mesos.internal.KillTaskMessage: Post http://127.0.0.1:5050/master/mesos.internal.KillTaskMessage: dial tcp 127.0.0.1:5050: getsockopt: connection refused '
I0827 13:37:17.088635    2418 scheduler.go:860] Aborting framework [&FrameworkID{Value:*20150803-161211-16777343-5050-19639-0000,XXX_unrecognized:[],}]
I0827 13:37:17.090779    2418 scheduler.go:815] Stopping the scheduler driver
I0827 13:37:17.090977    2418 messenger.go:166] Sending message mesos.internal.UnregisterFrameworkMessage to master@127.0.0.1:5050
```

**Comment by tsenart** [comment]
@jdef, @sttts: Isn't this a duplicate of #155?

**Comment by sttts** [comment]
They are certainly related in the sense that #155 shows the same fatal behavior at the end of the logs. But even when fixing #160, the blocking behavior of the http transport on failover in #155 will not be fixed. So I guess we are talking about two issues here.

**Comment by jdef** [comment]
If the master dies/fail over it's probably OK to lose most messages
in-flight: robust schedulers will implement reconciliation and re-send
important messages anyway. The driver should accommodate this line of
thinking instead of aborting.

**Comment by jdef** [comment]
status-update messages should be guaranteed delivery; everything else can be dropped

**Comment by jdef** [comment]
fixed by #221
